geode-redis/src/integrationTest/java/org/apache/geode/redis/PubSubIntegrationTest.java

```diff
@@ -253,11 +251,14 @@ public void testPatternSubscribe() {
 
     waitFor(() -> mockSubscriber.getSubscribedChannels() == 1);
 
-    Long result = publisher.publish("salutations", "hello");
+    String message = "hello-" + System.currentTimeMillis();
+
+    Long result = publisher.publish("salutations", message);
     assertThat(result).isEqualTo(1);
 
     assertThat(mockSubscriber.getReceivedMessages()).isEmpty();
```

Comment:
It seems like this old assert that we have no receivedMessages should be AFTER the new await that we did receive a PMessage. I could be wrong about this but it seems like you should have the test wait for the event and then do the asserts

Reply:
There are actually 2 categories of received messages - one is as a result of a 'regular' subscription and one is as a result of a pattern subscription. They are both sent and processed separately. So it doesn't matter in which order these asserts are done in. In this case I just want to make sure I received 1 'pmessage' and did not receive any other messages.

Reply:
But if you made sure AFTER you had received the message you are expecting, you would give the test more of a chance to catch that it also received the unexpected message. I don't think you should wait for something you never expect to happen; that would cause all tests to take too long to run. I'm okay if you leave the test as is